liquidator/src/liquidate.rs

```diff
@@ -572,18 +572,26 @@ impl<'a> LiquidateHelper<'a> {
         //
         // TODO: All these close ix could be in one transaction.
         if let Some(txsig) = self.perp_close_orders().await? {
-            return Ok(txsig);
+            return Ok(Some(txsig));
         }
         if let Some(txsig) = self.serum3_close_orders().await? {
-            return Ok(txsig);
+            return Ok(Some(txsig));
+        }
+
+        if self.health_cache.has_phase1_liquidatable() {
```

Comment:
is the local copy of account updated between cancelling perp/serum orders and here? if not this would always return true no?

Reply:
This currently runs one step of the liquidation and then returns. So yeah, the next time this function is called, it'll work from an updated account state.